backend/api/src/api/v1/gql/error.rs

```diff
@@ -0,0 +1,213 @@
+use std::{
+    fmt::{Display, Formatter},
+    panic::Location,
+};
+
+use async_graphql::ErrorExtensions;
+
+pub type Result<T, E = GqlErrorInterface> = std::result::Result<T, E>;
+
+pub struct GqlErrorInterface {
+    kind: GqlError,
```

Comment:
This entire file is a trick to allow the `?` operator to work. This is technically the error type, but if I were to name the other one `GqlErrorKind` you would only see that one littered everywhere. Since that is the only one we actually publicly reference.